jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/OSGiDeployableBundleMetadata.java

```diff
@@ -115,139 +130,90 @@ private static String getContextPath(Bundle bundle)
+     * @return a Resource representing the bundle's installed location
+     */
+    public Resource getBundleResource(ResourceFactory resourceFactory) throws Exception
     {
-        _contextPath = contextPath;
-    }
+        String bundleOverrideLocation = bundle.getHeaders().get(OSGiWebappConstants.JETTY_BUNDLE_INSTALL_LOCATION_OVERRIDE);
+        File bundleLocation = BundleFileLocatorHelperFactory.getFactory().getHelper().getBundleInstallLocation(bundle);
+        File root = (bundleOverrideLocation == null ? bundleLocation : new File(bundleOverrideLocation));
+        // Fix some osgiPaths.get( locations which point to an archive, but that doesn't end in .jar
+        URL url = BundleFileLocatorHelperFactory.getFactory().getHelper().getLocalURL(URIUtil.correctURI(root.toURI()).toURL());
 
-    public String getBundleSymbolicName()
-    {
-        return _bundle.getSymbolicName();
+        return resourceFactory.newResource(url);
     }
 
-    public String getBundleVersionAsString()
+    public String getContextPath()
     {
-        if (_bundle.getVersion() == null)
-            return null;
-        return _bundle.getVersion().toString();
+        return contextPath;
     }
 
-    public Bundle getBundle()
+    public String getID()
     {
-        return _bundle;
+        return bundle.getSymbolicName();
     }
 
-    public void setRegistration(ServiceRegistration registration)
+    public Path getPath()
     {
-        _registration = registration;
+        return bundlePath;
     }
 
-    public ServiceRegistration getRegistration()
+    public String getPathToResourceBase()
     {
-        return _registration;
+        return pathToResourceBase;
     }
 
-    /**
-     * Register the Jetty deployed context/webapp as a service, as
-     * according to the OSGi Web Application Specification.
-     */
-    public void registerAsOSGiService() throws Exception
+    public void setPathToResourceBase(String resourceBase)
     {
-        if (_registration == null)
-        {
-            Dictionary<String, String> properties = new Hashtable<String, String>();
-            properties.put(OSGiWebappConstants.WATERMARK, OSGiWebappConstants.WATERMARK);
-            if (getBundleSymbolicName() != null)
-                properties.put(OSGiWebappConstants.OSGI_WEB_SYMBOLICNAME, getBundleSymbolicName());
-            if (getBundleVersionAsString() != null)
-                properties.put(OSGiWebappConstants.OSGI_WEB_VERSION, getBundleVersionAsString());
-            properties.put(OSGiWebappConstants.OSGI_WEB_CONTEXTPATH, getContextPath());
-            ServiceRegistration rego = FrameworkUtil.getBundle(this.getClass()).getBundleContext().registerService(ContextHandler.class.getName(), getContextHandler(), properties);
-            setRegistration(rego);
-        }
+        this.pathToResourceBase = resourceBase;
     }
 
-    protected void deregisterAsOSGiService() throws Exception
+    public Properties getProperties()
```

Comment:
Properties are never set, we can delete the field and this method.

Reply:
The properties are set/put here ...

https://github.com/jetty/jetty.project/blob/de22789708692f773bc5c13c9e67ef1773a42966/jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/BundleContextProvider.java#L187-L193